lib/keystore.cc

```diff
@@ -97,6 +114,13 @@ static rpmRC write_key_to_disk(rpmPubkey key, string & dir, string & filename, i
 }
 
 
+static rpmRC delete_file_store(std::string path)
+{
+    fs::remove_all(path);
```

Comment:
Oh, one final thing. Sorry for missing earlier, this is belongs to a whole new class of issues to watch out for:
This can throw an exception that aborts the program whereas we should just return a success/failure error code.
So this needs to either use the non-throwing version, or catch possible exceptions.

Reply:
catching  `std::filesystem::filesystem_error` here is kinda straight forward. I wonder if we need a general policy on `std::bad_alloc`? We have the special alloc functions all over the code. I guess we want to keep that behaviour overall.

Reply:
Oh, the filesystem part is all I meant, sorry I guess I wasn't clear on that.

bad_alloc we should just let through everywhere - the rpm policy has always been to just die on alloc failures because continuing is going to be futile.

Reply:
So yeah just drop the bad_alloc check. We should document this someplace... okay, CONTRIBUTING has a section where this naturally falls into.

Reply:
stackoverflow agrees: https://stackoverflow.com/questions/9456728/how-to-deal-with-bad-alloc-in-c

> In general you cannot, and should not try, to respond to this error. bad_alloc indicates that a resource cannot be allocated because not enough memory is available. In most scenarios your program cannot hope to cope with that, and terminating soon is the only meaningful behaviour.

Reply:
There goes: https://github.com/rpm-software-management/rpm/commit/4c144d0825f975106e4ae1ae74c265b66391007e